tiledb/api/src/stats.rs

```diff
@@ -93,20 +102,22 @@ pub fn dump() -> TileDBResult<Option<String>> {
 }
 
 #[cfg(feature = "serde")]
-pub fn dump_json() -> TileDBResult<Option<Vec<Metrics>>> {
+mod __serde {
     use anyhow::anyhow;
 
-    if let Some(dump) = dump()? {
-        let datas: Vec<Metrics> = serde_json::from_str::<Vec<Metrics>>(
-            dump.as_str(),
-        )
-        .map_err(|e| {
-            Error::Deserialization(
-                format!("Failed to deserialize stats JSON value {}", dump),
-                anyhow!(e),
-            )
-        })?;
-        return Ok(Some(datas));
+    use super::*;
+
+    #[cfg(feature = "serde")]
+    pub fn dump_json() -> Result<Option<Vec<Metrics>>, Error> {
+        if let Some(dump) = dump()? {
+            Ok(serde_json::from_str::<Vec<Metrics>>(dump.as_str())
+                .map(Some)
+                .map_err(|e| Error::ToJson(anyhow!(e)))?)
+        } else {
+            Ok(None)
+        }
     }
-    Ok(None)
 }
+
+#[cfg(feature = "serde")]
+pub use __serde::*;
```

Comment:
I assume there's a good reason for doing this the way you've got it, but I'm not seeing it off the top of my head. Can you drop a comment explaining the setup/motivation?

Reply:
I think this was really just to limit the number of `#[cfg(feature = "serde")]` but given that there isn't much in the module that seems a little silly.